autopilot/interface.go

```diff
@@ -129,6 +145,16 @@ type AttachmentHeuristic interface {
 	Select(self *btcec.PublicKey, graph ChannelGraph,
 		amtToUse btcutil.Amount, numNewChans uint32,
 		skipNodes map[NodeID]struct{}) ([]AttachmentDirective, error)
+
+	// NodeScores is a method that given the current channel graph, current
+	// set of local channels and funds available, scores the given nodes
+	// according the the preference of opening a channel with them. The
+	// returned channel candidates maps the NodeID to a score and a channel
+	// size. The scores will be in the range [0, M], where 0 indicates no
+	// improvement in connectivity if a channel is opened to this node,
+	// while M is the maximum possible improvement in connectivity.
+	NodeScores(g ChannelGraph, chans []Channel, fundsAvailable btcutil.Amount,
```

Comment:
Depending on how the future heuristics specify this score, we may want to add a degree of normalization across all the scores. Otherwise, we may be comparing something like the in-degree of nodes to the aggregate amount that we've routed through them in the past (just an example).